### PR TITLE
Implementing phabricator task T2549

### PR DESCRIPTION
**Pull request by sau226**
I am implementing miraheze phab task 2549. Please do not merge unless all your reviews have been done (obviously). Thank you

### CODE REVIEW

File: LocalSettings.php [hunk 1]
```diff
@@ -6918,7 +6971,7 @@
 	'wmgPrivateWiki' => array(
 		'default' => false,
 	),
-);
+)
```
Comment (Reception123) [inline LocalSettings.php]: it should be ),

File: LocalWiki.php [hunk 2]
```diff
@@ -200,6 +200,16 @@ function onClosedSiteNoticeAfter( &$siteNotice, $skin ) {
 	$wgGroupPermissions['*']['createtalk'] = false;
 }
 
+if ( $wgDBname === 'sau226wiki' ) {
+	$wgGroupPermissions['sysop']['abusefilter-modify-restricted'] = false;
+	$wgGroupPermissions['sysop']['editinterface'] = false;
+	$wgGroupPermissions['sysop']['globalblock-whitelist'] = false;
+	$wgGroupPermissions['sysop']['ipblock-exempt'] = false;
+	$wgGroupPermissions['sysop']['nuke'] = false;
+	$wgGroupPermissions['bureaucrat']['managewiki'] = false;
```
Comment (Reception123) [inline LocalWiki.php]: Please do this with wmgManageWikiGroup rather than like this.

File: LocalWiki.php [hunk 3]
```diff
@@ -200,6 +200,16 @@ function onClosedSiteNoticeAfter( &$siteNotice, $skin ) {
...
+	$wgGroupsRemoveFromSelf['consul'][] = 'consul';
```
Comment (Reception123) [inline LocalWiki.php]: Please do this in LocalSettings.php as well.

File: LocalSettings.php (hunk 1 above)
Comment (sau226) [reply LocalSettings.php]: Done

File: LocalWiki.php (hunk 2 above)
Comment (sau226) [reply LocalWiki.php]: That's fine by me. However you appear to use manual overrides for other wikis so I need clarification for this. I have dealt with managewiki by LocalSettings.php and have removed them from LocalWiki.php

File: LocalWiki.php (hunk 3 above)
Comment (sau226) [reply LocalWiki.php]: Sorry but I need a bit of assistance regarding what code to use for wgGroupsRemoveFromSelf . Please also see my comment above

**Comment by sau226** [comment]
Travis CI appears to have a problem with the review. Naturally you probably can override it as repo owner but you may want to double check that

File: LocalWiki.php (hunk 2 above)
Comment (sau226) [reply LocalWiki.php]: I have done it with the intention that if it doesn't work I will define manual overrides in LocalWiki.php if LocalSettings.php doesn't work

**Comment by sau226** [comment]
All requested changes implemented. Please review/merge. Thanks

File: LocalSettings.php [hunk 4]
```diff
@@ -5449,6 +5486,7 @@
 	),
 	'wgGroupsRemoveFromSelf' => array(
 		'default' => array(),
+		'sau226wiki' => [ 'consul' => [ 'consul' ] ], //T2549
```
Comment (Reception123) [inline LocalSettings.php]: Please use ) arrays

Comment (sau226) [reply LocalSettings.php]: I hope arrays like the ones below work. I have used a modified copy of an array below. Should hopefully work. Please re-review. Thank you very much

**Comment by sau226** [comment]
Also if you merge this will it take ~10 minutes for puppet or whatever sync tool like that you have to update the server? Or will it take less/more time?

File: LocalSettings.php [hunk 5]
```diff
@@ -6918,7 +6975,7 @@
 	'wmgPrivateWiki' => array(
 		'default' => false,
 	),
-);
+),
```
Comment (JohnFLewis) [inline LocalSettings.php]: Needs to be ); as it was before.

Comment (sau226) [reply LocalSettings.php]: Done

File: LocalSettings.php [hunk 6]
```diff
@@ -5449,6 +5486,10 @@
 	),
 	'wgGroupsRemoveFromSelf' => array(
 		'default' => array(),
+		'+sau226wiki' => array(
+			'consul' => array(
+				'consul',
+			),
```
Comment (Reception123) [inline LocalSettings.php]: You are missing a closing array here

Comment (sau226) [reply LocalSettings.php]: Done or will be done soon

**Comment by sau226** [comment]
Thank you all for being so patient and mentory. I will hopefully make my last commit in relation to this PR soon.

**Pull request merged**